### PR TITLE
Improve the notification logs when the user has no email address

### PR DESCRIPTION
**Pull request by 2403905**
## Description

Improve the notification logs when the user has no email address

## Related Issue

- Fixes [https://github.com/owncloud/ocis/issues/6855](https://github.com/owncloud/ocis/issues/6855)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: services/notifications/pkg/service/service.go [hunk 1]
```diff
@@ -126,6 +127,18 @@ func (s eventsNotifier) send(ctx context.Context, recipientList []*channels.Mess
 	}
 }
 
+func (s eventsNotifier) ensureGranteeList(ctx context.Context, executant, u *user.UserId, g *group.GroupId) []*user.User {
+	if granteeList, err := s.getGranteeList(ctx, executant, u, g); err != nil {
+		s.logger.Error().Err(err).Str("event", "ensureGranteeList").Msg("Could not get grantee list")
+		return nil
+	} else if len(granteeList) == 0 {
+		s.logger.Debug().Str("event", "ensureGranteeList").Msg("The grantee list is empty")
```
Comment (kobergj) [inline services/notifications/pkg/service/service.go:135]: Do we need this log? Getting an empty grantee list is nothing extraordinary

File: services/notifications/pkg/service/service.go [hunk 2]
```diff
@@ -126,6 +127,18 @@ func (s eventsNotifier) send(ctx context.Context, recipientList []*channels.Mess
...
+		return nil
+	} else {
```
Comment (kobergj) [inline services/notifications/pkg/service/service.go:137]: Please remove the last else and return `granteeList` in main thread

**Review by kobergj — CHANGES_REQUESTED** [review]

File: services/notifications/pkg/service/service.go (hunk 1 above)
Comment (2403905) [reply services/notifications/pkg/service/service.go:135]: The omitting of the email is not allowed in the UI. These logs should help us understand if something wrong

File: services/notifications/pkg/service/service.go (hunk 2 above)
Comment (2403905) [reply services/notifications/pkg/service/service.go:137]: ok

File: services/notifications/pkg/service/service.go (hunk 1 above)
Comment (2403905) [reply services/notifications/pkg/service/service.go]: We have other logs that tell us if the email is empty

**Review by kobergj — APPROVED** [review]

**Pull request merged**